src/models/logic/transaction.ts

```diff
@@ -488,6 +471,43 @@ export async function getTransactions(params: {
     }
 }
 
+type BlockId =
+    | {
+          type: "number";
+          blockNumber: number;
+      }
+    | {
+          type: "hash";
+          blockHash: H256;
+      };
+
+export async function getTransactionsOfBlock(params: {
+    page: number;
+    itemsPerPage: number;
+    blockId: BlockId;
+}) {
+    try {
+        const { page, itemsPerPage, blockId } = params;
+        return models.Transaction.findAll({
+            where:
+                blockId.type === "number"
+                    ? {
+                          blockNumber: blockId.blockNumber
+                      }
+                    : {
+                          blockHash: blockId.blockHash.value
```

Comment:
Finding blocks using the `blockHash` and order the results using `blockNumber` seems not using the DB index. Did you check it?

Reply:
What do you think about querying `blockNumber` using `blockHash`?

Reply:
I didn't check it since there is a certain limit for the # of txs in a block.

Do you mean fetching the `blockNumber` first, and then, use the `blockNumber` for the `findAll`?

Reply:
I realized that the `["blockNumber", "DESC"]` part is useless because all the fetched rows will have the same `blockNumber`.

I think there are 2 ways to utilize the index for this query. One is what you suggested and the other is changing the index from `["blockHash"]` to `["blockHash", "transactionIndex"]`

Reply:
I like the method I suggested. It will reduce DB overhead a little.
I think that both two methods are good.

Reply:
Agree. I will fix it as your suggestion.